### DISCUSSION TITLE
疑问，既然测速能够获得最快ip，为什么还要国内外分流查询？岂不是多此一举？

### DISCUSSION THREAD

**Discussion by 57382** [pymumu/smartdns#1415] [Q&A] [answered]
### _**对于开启了缓存功能后，缓存后的数据，无论使用哪种方式，其结果是最佳的，所以不建议关闭缓存。**_

查询国外域名使用国外DNS和让这个DNS通过代理远程查询，
查询国内域名用114。
这样避免国内域名被解析到国外ip，和避免污染。

可是，SmartDNS已经能够通过测速返回速度最快的域名ip，
分流是不是就没意义了？

**Comment by PikuZheng** [comment]
抗污染

**Reply by PikuZheng** [reply, marked as answer]
国外的测速不一定能得到正确结果

**Reply by 57382** [reply]
> 抗污染


@PikuZheng  谢谢答疑，我明白你的意思了

以访问abc.com为例，假设abc是国内网站，
假设abc最快ip是1234，还有一个慢ip5555，
假设114记录了1234，CF的1111记录了5555，
这时候只使用CF1111测速，得到5555，不是最快的，
虽然5555是测速结果当中最快的，更快的114带的1234没加入，
这才让5555拿了第一，为了抗污只用了国外DNS。

所以需要分流查询，除非不在意是不是真的最快。

**Reply by ListenV** [reply]
没看明白

**Reply by PikuZheng** [reply]
他说的我也没看明白。但是他自己想通了。

**Reply by ZqinKing** [reply]
> 抗污染

除了污染还有一个附加能力，就是国外组DOT或者DOH走代理时，上游DNS服务器根据出口IP地址，会提供最近的IP地址作为响应。
